Connectors/Autocad/Speckle.Connectors.AutocadShared/Operations/Receive/AutocadHostObjectBaseBuilder.cs

```diff
@@ -0,0 +1,288 @@
+using Autodesk.AutoCAD.DatabaseServices;
+using Speckle.Connectors.Autocad.HostApp;
+using Speckle.Connectors.Autocad.HostApp.Extensions;
+using Speckle.Connectors.Common.Builders;
+using Speckle.Connectors.Common.Conversion;
+using Speckle.Connectors.Common.Extensions;
+using Speckle.Connectors.Common.Operations;
+using Speckle.Connectors.Common.Operations.Receive;
+using Speckle.Converters.Common;
+using Speckle.Sdk.Common;
+using Speckle.Sdk.Dependencies;
+using Speckle.Sdk.Models;
+using Speckle.Sdk.Models.Collections;
+using Speckle.Sdk.Models.Instances;
+using AutocadColor = Autodesk.AutoCAD.Colors.Color;
+
+namespace Speckle.Connectors.Autocad.Operations.Receive;
+
+/// <summary>
+/// <para>Base class for AutoCAD host object builders. Expects to be a scoped dependency per receive operation.</para>
+/// </summary>
+public abstract class AutocadHostObjectBaseBuilder : IHostObjectBuilder
+{
+  private readonly IRootToHostConverter _converter;
+  private readonly AutocadLayerBaker _layerBaker;
+  private readonly AutocadGroupBaker _groupBaker;
+  private readonly AutocadInstanceBaker _instanceBaker;
+  private readonly IAutocadMaterialBaker _materialBaker;
+  private readonly IAutocadColorBaker _colorBaker;
+  private readonly AutocadContext _autocadContext;
+  private readonly RootObjectUnpacker _rootObjectUnpacker;
+  private readonly IReceiveConversionHandler _conversionHandler;
+
+  protected AutocadHostObjectBaseBuilder(
+    IRootToHostConverter converter,
+    AutocadLayerBaker layerBaker,
+    AutocadGroupBaker groupBaker,
+    AutocadInstanceBaker instanceBaker,
+    IAutocadMaterialBaker materialBaker,
+    IAutocadColorBaker colorBaker,
+    AutocadContext autocadContext,
+    RootObjectUnpacker rootObjectUnpacker,
+    IReceiveConversionHandler conversionHandler
+  )
+  {
+    _converter = converter;
+    _layerBaker = layerBaker;
+    _groupBaker = groupBaker;
+    _instanceBaker = instanceBaker;
+    _materialBaker = materialBaker;
+    _colorBaker = colorBaker;
+    _autocadContext = autocadContext;
+    _rootObjectUnpacker = rootObjectUnpacker;
+    _conversionHandler = conversionHandler;
+  }
+
+  public Task<HostObjectBuilderResult> Build(
+    Base rootObject,
+    string projectName,
+    string modelName,
+    IProgress<CardProgress> onOperationProgressed,
+    CancellationToken cancellationToken
+  )
+  {
+    // Prompt the UI conversion started. Progress bar will swoosh.
+    onOperationProgressed.Report(new("Converting", null));
+
+    // Layer filter for received commit with project and model name
+    _layerBaker.CreateLayerFilter(projectName, modelName);
+
+    // 0 - Clean then Rock n Roll!
+    string baseLayerPrefix = _autocadContext.RemoveInvalidChars($"SPK-{projectName}-{modelName}-");
+    PreReceiveDeepClean(baseLayerPrefix);
```

Comment:
This is the method that should be overridden in civil, with the property set purge call added